tests/sentry/workflow_engine/handlers/condition/test_group_state_handlers.py

```diff
@@ -207,3 +208,81 @@ def test_dual_write(self):
         assert dc.comparison is True
         assert dc.condition_result is True
         assert dc.condition_group == dcg
+
+
+class TestNewHighPriorityIssueCondition(ConditionTestCase):
+    condition = Condition.NEW_HIGH_PRIORITY_ISSUE
+    rule_cls = NewHighPriorityIssueCondition
+    payload = {"id": NewHighPriorityIssueCondition.id}
+
+    def setUp(self):
+        super().setUp()
+        self.job = WorkflowJob(
+            {
+                "event": self.group_event,
+                "group_state": GroupState(
+                    {
+                        "id": 1,
+                        "is_regression": True,
+                        "is_new": True,
+                        "is_new_group_environment": True,
+                    }
+                ),
+                "workflow": Workflow(environment_id=1),
+            }
+        )
+        self.dc = self.create_data_condition(
+            type=self.condition,
+            comparison=True,
+            condition_result=True,
+        )
+
+    def test_with_high_priority_alerts(self):
+        self.project.flags.has_high_priority_alerts = True
+        self.project.save()
+
+        # This will only pass for new issues
+        self.group_event.group.update(priority=PriorityLevel.HIGH)
+        self.job.update({"group_state": {"is_new_group_environment": True}})
```

Comment:
i think this needs to be `self.job["group_state"]["is_new_group_environment"] = True`. you can call `update()` because `job` is a dictionary, but mypy is complaining that `group_state` is missing some keys.

we should call `update()` for self.group_event.group because it's a Django model and we want to save the changes to the db